File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/credential/S3InstanceProfileCredentialConfig.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3.credential;

import com.amazonaws.auth.AWSCredentialsProvider;
import com.amazonaws.auth.InstanceProfileCredentialsProvider;

public class S3InstanceProfileCredentialConfig implements S3CredentialConfig {

  @Override
  public S3CredentialType getCredentialType() {
    return S3CredentialType.DEFAULT_PROFILE;
  }

  @Override
  public AWSCredentialsProvider getS3CredentialsProvider() {
    return new InstanceProfileCredentialsProvider(false);
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/credential/S3InstanceProfileCredentialConfig.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3.credential

import com.amazonaws.auth.AWSCredentialsProvider
import com.amazonaws.auth.InstanceProfileCredentialsProvider

class S3InstanceProfileCredentialConfig : S3CredentialConfig {
    override val credentialType: S3CredentialType
        get() = S3CredentialType.DEFAULT_PROFILE

    override val s3CredentialsProvider: AWSCredentialsProvider
        get() = InstanceProfileCredentialsProvider(false)
}
